File: programs/h-index-ii.go
package main

import "fmt"

func hIndex(citations []int) int {
	result := 0
	l := len(citations)
	for i := range citations {
		if citations[i] >= l-i {
			result = l - i
			break
		}
	}
	return result
}

func main() {
	test := []int{11, 15}
	fmt.Println(hIndex(test))
}

File: programs/h-index-ii.js
var hIndex = function(citations) {
  var result = 0, len = citations.length;
  for (var i = 0; i < len; ++i) {
    if (citations[i] >= len - i) {
      result = len - i;
      break;
    }
  }
  return result;
};

console.log(hIndex([11, 15]));
console.log(hIndex([0]));
console.log(hIndex([100]));
